src/MinimalDelegation.sol

```diff
@@ -73,6 +80,28 @@ contract MinimalDelegation is IERC7821, IKeyManagement, ERC1271, EIP712 {
         return mode.isBatchedCall() || mode.supportsOpData();
     }
 
+    /// @inheritdoc IAccount
+    function validateUserOp(PackedUserOperation calldata userOp, bytes32 userOpHash, uint256 missingAccountFunds)
+        external
+        returns (uint256 validationData)
+    {
+        if (msg.sender != ENTRY_POINT) revert NotEntryPoint();
+
+        // https://github.com/coinbase/smart-wallet/blob/main/src/CoinbaseSmartWallet.sol#L100
+        assembly ("memory-safe") {
+            if missingAccountFunds {
+                // Ignore failure (it's EntryPoint's job to verify, not the account's).
+                pop(call(gas(), caller(), missingAccountFunds, codesize(), 0x00, codesize(), 0x00))
+            }
+        }
+
+        if (_isValidSignature(userOpHash, userOp.signature)) {
```

Comment:
Nice! I guess we don't need to re-hash the hash here like regular 1271 since the userOp includes the current address in the sender field, should be replay safe across accounts